src/Mirza/BusinessRegistry/Main.hs

```diff
@@ -317,6 +322,48 @@ printCredentials user = do
   putStrLn $ "Password: " <> show (newUserPassword user)
 
 
+--------------------------------------------------------------------------------
+-- Bootstrap Command
+--------------------------------------------------------------------------------
+
+-- This command is a bit of a hack. We need it so that we can noninteractively
+-- add a user to the database so that this user can authenticate over the API
+-- and add other users and businesses into the database. The fact that this user
+-- doesn't take proper credentials suggests that our user model is wrong
+-- (incomplete)... i.e. we need users that aren't associated with businesses,
+-- but we need to do much more work here when we deal with permssions in general.
+runBootstrap :: ServerOptionsBR -> EmailAddress -> Text -> GS1CompanyPrefix -> IO ()
+runBootstrap opts email password companyPrefix = do
+  let newBusiness = bootstrapBusiness companyPrefix
+  let newUser = bootstrapUser email password companyPrefix
+
+  ctx        <- initBRContext opts
+  -- We ignore the business insert result, because the business may already
+  -- exist and so for now we just try best effort on the user because thats what
+  -- we care about. Can always improve the check and error handling here later
+  -- if we need to improve the reliability or error reporting.
+  _result    <- runAppM @_ @BRError ctx $ addBusiness newBusiness
+  userResult <- runAppM @_ @BRError ctx $ addUser newUser
+  either (print @BRError) print userResult
+
+  where
+    bootstrapBusiness :: GS1CompanyPrefix -> NewBusiness
+    bootstrapBusiness prefix = do
+      let newBusinessGS1CompanyPrefix = prefix
+      let newBusinessName             = "Bootstrapped Business"
+      NewBusiness{..}
+
+    bootstrapUser :: EmailAddress -> Text -> GS1CompanyPrefix -> NewUser
+    bootstrapUser userEmail userPassword companyPrefix = do
```

Comment:
This is overshadowing the definition at line 336. Just change it to `pfx` or something :slightly_smiling_face:

Reply:
Thanks for picking this up, have fixed 080e31c7359946056124c454b2f6951fc1f3e6d7.